batcher/aligned-batcher/src/lib.rs

```diff
@@ -261,17 +263,48 @@ impl Batcher {
     }
 
     pub async fn listen_new_blocks(self: Arc<Self>) -> Result<(), BatcherError> {
-        let mut stream = self
-            .eth_ws_provider
-            .subscribe_blocks()
-            .await
-            .map_err(|e| BatcherError::EthereumSubscriptionError(e.to_string()))?;
+        retry_function(
+            || {
+                let app = self.clone();
+                async move { app.listen_new_blocks_retryable().await }
+            },
+            DEFAULT_MIN_DELAY,
+            DEFAULT_FACTOR,
+            DEFAULT_MAX_TIMES,
```

Comment:
`DEFAULT_MAX_TIMES` is constant across the whole execution of the batcher. It would be good to restart the max times upon a correct response. That or increase the value.

Reply:
See if this applies to other points.

Reply:
I think that by using a large number of retries we avoid this issue